Examples/Io/Root/include/ActsExamples/Io/Root/RootNTupleReader.hpp

```diff
@@ -0,0 +1,142 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Propagator/MaterialInteractor.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/Framework/IReader.hpp"
+#include "ActsExamples/Framework/IService.hpp"
+#include "ActsExamples/Framework/ProcessCode.hpp"
+
+#include <mutex>
+#include <vector>
+
+class TChain;
+
+namespace ActsExamples {
+
+class RootNTupleReader : public ActsExamples::IReader {
+ public:
+  /// @brief The nested configuration struct
+  struct Config {
+    // name of the input tree
+    std::string inputTreeName;
+    // The name of the input file
+    std::string inputFilePath;
+
+    std::string outputTrackParameters = "nTupleTrackParameters";
+    std::string outputTruthVtxParameters = "nTupleTruthVtxParameters";
+    std::string outputRecoVtxParameters = "nTupleRecoVtxParameters";
+    std::string outputBranchPointerWrapper = "nTupleBranchPointerWrapper";
+    std::string outputBeamspotConstraint = "beamspotConstraint";
+  };
+
+  struct BranchPointerWrapper {
+    std::vector<float> *m_track_d0;
```

Comment:
As mentioned in the other comment, I think these vectors can simply live on the stack.

Reply:
hmmm not sure but I think I saw this a few times with root. the pointer is later passed by reference to root so it can be redirected to the data in the event.

I think we also have this with other root code in acts :thinking: